src/utils/error.ts

```diff
@@ -0,0 +1,31 @@
+import { NodeBundlerName } from '../runtimes/node/bundlers'
+import { RuntimeName } from '../runtimes/runtime'
+
+interface CustomErrorInfo {
+  type: 'functionsBundling'
+  location: {
+    functionName: string
+    runtime: RuntimeName
+    bundler?: NodeBundlerName
+  }
+}
+
+export class FunctionBundlingUserError extends Error {
+  customErrorInfo: CustomErrorInfo
+
+  constructor(
+    message: string,
+    customErrorInfo: {
+      functionName: string
+      runtime: RuntimeName
+      bundler?: NodeBundlerName
+    },
+  ) {
+    super(message)
+
+    Object.setPrototypeOf(this, new.target.prototype)
+    this.name = 'FunctionBundlingUserError'
```

Comment:
These two lines are necessary because typescript does not support extending builtins completely
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-2.html#support-for-newtarget